airflow/models/taskinstance.py

```diff
@@ -2142,7 +2152,5 @@ def construct_task_instance(self, session=None, lock_for_update=False) -> TaskIn
 if STATICA_HACK:  # pragma: no cover
 
     from airflow.job.base_job import BaseJob
-    from airflow.models.dagrun import DagRun
 
-    TaskInstance.dag_run = relationship(DagRun)
```

Comment:
The primary reason for having this was to help with code-completion, so maybe we should keep the relationship on dagrun and change `backref` to `back_populates` as suggested